### PR TITLE
[lgwebos] Added rcButton channel

### PR DESCRIPTION
**Pull request by bbrodt**
This PR adds a new channel ("IRCode") to the LG WebOS binding, which simulates some of the button presses on the LG infrared remote control. Inspiration for this enhancement came [from the discussion here.](https://community.openhab.org/t/lgwebos-binding-sendbutton-question/85872/8)

Here is a list of IR codes that I have found to work. Note that this list has been compiled mostly through trial and error, and I may have missed some of the buttons. I have only been able to test this on an LGUJ657A TV so your mileage may vary.

Code String | Description
-- | --
LEFT | Left button in cursor control group
RIGHT | Right button in cursor control group
UP | Up button in cursor control group
DOWN | Down button in cursor control group
ENTER | "OK" button in the center of the cursor control group
BACK | "BACK" button
EXIT | "EXIT" button
0-9 | Number buttons
HOME | "HOME" button
RED | "RED" button
GREEN | "GREEN" button
YELLOW | "YELLOW" button
BLUE | "BLUE" button
PLAY | "PLAY" button
PAUSE | "PAUSE" button
STOP | "STOP" button

A sample HABPanel remote control widget can be found [in this github repository.](https://github.com/bbrodt/openhab2-misc)

### CODE REVIEW

File: bundles/org.openhab.binding.lgwebos/src/main/resources/ESH-INF/i18n/lgwebos.properties [hunk 1]
```diff
@@ -38,3 +38,8 @@ actionIncreaseChannelDesc=TV will switch one channel up in the current channel l
 
 actionDecreaseChannelLabel=Channel Down
 actionDecreaseChannelDesc=TV will switch one channel down in the current channel list.
+
+actionShowRCButtonLabel=Remote Control button press
```
Comment (sprehn) [inline bundles/org.openhab.binding.lgwebos/src/main/resources/ESH-INF/i18n/lgwebos.properties]: You have added the labels to support a new action, but I miss changes to class LGWebOSAction as part of this PR

Comment (bbrodt) [reply bundles/org.openhab.binding.lgwebos/src/main/resources/ESH-INF/i18n/lgwebos.properties]: Sorry, missed that. I'm on it...

Comment (bbrodt) [reply bundles/org.openhab.binding.lgwebos/src/main/resources/ESH-INF/i18n/lgwebos.properties]: OK, please review new commit.

File: bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/action/LGWebOSActions.java [hunk 2]
```diff
@@ -237,6 +237,12 @@ public void decreaseChannel() {
         getConnectedSocket().ifPresent(control -> control.channelDown(createResponseListener()));
     }
 
+    @RuleAction(label = "@text/actionSendRCButtonLabel", description = "@text/actionSendRCButtonDesc")
+    public void sendRCButton(
```
Comment (sprehn) [inline bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/action/LGWebOSActions.java]: please also add a corresponding static method below 
// delegation methods for "legacy" rule support

Comment (bbrodt) [reply bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/action/LGWebOSActions.java]: OK done. Please review ... again :)

**Comment by rufinus** [comment]
@bbrodt  would you mind releasing a test version of the plugin jar file for me to test?

File: bundles/org.openhab.binding.lgwebos/src/main/resources/ESH-INF/i18n/lgwebos.properties (hunk 1 above)
Comment (sprehn) [reply bundles/org.openhab.binding.lgwebos/src/main/resources/ESH-INF/i18n/lgwebos.properties]: looks good to me

**Comment by sprehn** [comment]
Looks good to me. Thank you @bbrodt for this contribution and Merry Christmas!

**Comment by bbrodt** [comment]
> @bbrodt would you mind releasing a test version of the plugin jar file for me to test?

The jars can be found here:
https://drive.google.com/drive/folders/1mynG60cnLLQnbNEj1bxcMY8KNMGv1G8y?usp=sharing

**Comment by rufinus** [comment]
@bbrodt  Thanks! - Tested it for navigating in the Netflix app - works great!

**Comment by J-N-K** [comment]
Could you please rebase and resolve merge conflicts? Thanks.

**Comment by bbrodt** [comment]
Hi @sprehn  and Happy New Year! :)
I tried rebasing, but there are waaaaay too many conflicts for my comfort level. I'm going to create a new fork, add my changes, and create a new PR.

I'm still a bit confused though - should I be using the default branch (2.5.x) from openhab-addons or the master branch, or should I create a feature branch?

Thanks!
Bob

**Comment by sprehn** [comment]
> I'm still a bit confused though - should I be using the default branch (2.5.x) from openhab-addons or the master branch, or should I create a feature branch?

I suppose you have a forked version of openhab-addons. Rebase master in your fork to the latest version from the official openhab repo. Create a feature branch in your fork. From that feature branch you create the pull request to openhab master.

**Comment by bbrodt** [comment]
Creating a new pull request (6891) based on the most recent fork of origin 2.5.x branch

**Pull request closed without merging**